kratos/geometries/coupling_geometry.h

```diff
@@ -0,0 +1,349 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Tobias Teschemacher
+//  Contributors:    Pooyan Dadvand
+//                   Philipp Bucher
+//
+
+#if !defined(KRATOS_COUPLING_GEOMETRY_H_INCLUDED )
+#define  KRATOS_COUPLING_GEOMETRY_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "geometry.h"
+
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+
+/**
+ * @class CouplingGeometry
+ * @ingroup KratosCore
+ * @brief The CouplingGeometry can connect different geometries, those
+ can be of different entities but have to coincide with the dimension.
+ */
+template<class TPointType> class CouplingGeometry
+    : public Geometry<TPointType>
+{
+public:
+    ///@}
+    ///@name Type Definitions
+    ///@{
+
+    /**
+     * Geometry as base class.
+     */
+    typedef Geometry<TPointType> BaseType;
+    typedef Geometry<TPointType> GeometryType;
+
+    typedef typename GeometryType::Pointer GeometryPointer;
+
+    /**
+     * Pointer definition of CouplingGeometry
+     */
+    KRATOS_CLASS_POINTER_DEFINITION( CouplingGeometry );
+
+    typedef TPointType PointType;
+
+    typedef typename BaseType::IndexType IndexType;
+    typedef typename BaseType::SizeType SizeType;
+
+    typedef typename BaseType::PointsArrayType PointsArrayType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor for coupling one master to one slave geometry.
+    CouplingGeometry(
+        GeometryPointer pMasterGeometry,
+        GeometryPointer pSlaveGeometry)
+        : BaseType(PointsArrayType(), &(pMasterGeometry->GetGeometryData()))
+    {
+        KRATOS_DEBUG_ERROR_IF(pMasterGeometry->Dimension() != pSlaveGeometry->Dimension())
+            << "Geometries of different dimensional size!" << std::endl;
```

Comment:
Why the geometries should have the same dimension? I understand that this would be your case but in general one may tie a cross section of a bridge to a point. Isn't it?

Reply:
Yes. But Dimension in this case is then a point on the bridge with the point. So local and working space dimension don't have to coincide. But the dimension has to because otherwise integration cannot be done correctly.

Reply:
I'm not sure if I got your rational. My idea is that this should be checked by Element/Condition who applies the kinematic not the geometry.

Reply:
Well, I wouldn't say that there it is necessary. It is definiteley an issue about who controls what. But from a physical point of view that is what matters.